btrfs headers issue with containerd for RHEL based distros

Same as https://github.com/docker/packaging/pull/69#issuecomment-1374731078 (fyi @neersighted)

https://github.com/docker/packaging/actions/runs/4158235007/jobs/7193316373#step:7:655

```
#41 52.20 + bin/containerd
#41 52.20 go build  -gcflags=-trimpath=/go/src -buildmode=pie  -o bin/containerd -ldflags '-X github.com/containerd/containerd/version.Version=v0.0.0-20230212032334-040fcf8 -X github.com/containerd/containerd/version.Revision=040fcf85f0a7d88ef3d55720d9821c9cb127c81c -X github.com/containerd/containerd/version.Package=containerd.io -s -w ' -tags "urfave_cli_no_docs"  ./cmd/containerd
#41 81.49 # github.com/containerd/btrfs/v2
#41 81.49 In file included from vendor/github.com/containerd/btrfs/v2/btrfs.go:21:0:
#41 81.49 ./btrfs.h:19:2: error: #error "Headers from kernel >= 4.12 are required on compilation time (not on run time)"
#41 81.49  #error "Headers from kernel >= 4.12 are required on compilation time (not on run time)"
#41 81.49   ^
#41 81.49 In file included from vendor/github.com/containerd/btrfs/v2/btrfs.go:21:0:
#41 81.49 ./btrfs.h:22:30: fatal error: linux/btrfs_tree.h: No such file or directory
#41 81.49  #include <linux/btrfs_tree.h>
```

--- a/pkg/containerd/Dockerfile
+++ b/pkg/containerd/Dockerfile
@@ -210,7 +210,7 @@ ARG NIGHTLY_BUILD
 ARG RUNC_REF
 WORKDIR /build
 ARG TARGETPLATFORM
-RUN xx-apt-get install -y libbtrfs-dev libseccomp-dev btrfs-progs gcc
+RUN xx-apt-get install -y libseccomp-dev gcc
 RUN --mount=type=bind,source=scripts/pkg-static-build.sh,target=/usr/local/bin/pkg-static-build \
     --mount=type=bind,from=common-scripts,source=gen-ver.sh,target=/usr/local/bin/gen-ver \
     --mount=type=bind,from=common-scripts,source=fix-cc.sh,target=/usr/local/bin/fix-cc \

--- a/pkg/containerd/deb/control
+++ b/pkg/containerd/deb/control
@@ -2,8 +2,7 @@ Source: containerd.io
 Section: devel
 Priority: optional
 Maintainer: Containerd team <help@containerd.io>
-Build-Depends: libbtrfs-dev | btrfs-tools,
-               debhelper (>= 10~) | dh-systemd,
+Build-Depends: debhelper (>= 10~) | dh-systemd,
                pkg-config,
                libseccomp-dev
 Standards-Version: 4.1.4

--- a/pkg/containerd/rpm/containerd.spec
+++ b/pkg/containerd/rpm/containerd.spec
@@ -69,16 +69,6 @@ BuildRequires: libtool-ltdl-devel
 BuildRequires: systemd
 BuildRequires: libseccomp-devel
 
-%if %{undefined rhel} || 0%{?rhel} < 8
-%if %{defined suse_version}
-# SUSE flavors
-BuildRequires: libbtrfs-devel
-%else
-# Fedora / others, and CentOS/RHEL < 8
-BuildRequires: btrfs-progs-devel
-%endif
-%endif
-
 %{?systemd_requires}
 
 %description
@@ -108,14 +98,7 @@ cd %{_topdir}/BUILD/
 %build
 cd %{_topdir}/BUILD
 GO111MODULE=auto make man
-
-BUILDTAGS=""
-%if %{defined rhel} && 0%{?rhel} >= 8
-# btrfs support was removed in CentOS/RHEL 8
-BUILDTAGS="${BUILDTAGS} no_btrfs"
-%endif
-
-GO111MODULE=auto make -C /go/src/%{import_path} VERSION=%{_origversion} REVISION=%{_commit} PACKAGE=%{getenv:PKG_NAME} BUILDTAGS="${BUILDTAGS}"
+GO111MODULE=auto make -C /go/src/%{import_path} VERSION=%{_origversion} REVISION=%{_commit} PACKAGE=%{getenv:PKG_NAME} BUILDTAGS="%{getenv:BUILDTAGS}"
 
 # Remove containerd-stress, as we're not shipping it as part of the packages
 rm -f bin/containerd-stress

--- a/pkg/containerd/scripts/pkg-rpm-build.sh
+++ b/pkg/containerd/scripts/pkg-rpm-build.sh
@@ -66,6 +66,12 @@ if [ -n "$(xx-info variant)" ]; then
   pkgoutput="${pkgoutput}/$(xx-info variant)"
 fi
 
+case "$PKG_RELEASE" in
+  centos*|oraclelinux*)
+    export BUILDTAGS="no_btrfs $BUILDTAGS"
+    ;;
+esac
+
 set -x
 
 sed 's#/usr/local/bin/containerd#/usr/bin/containerd#g' "${SRCDIR}/containerd.service" > /root/rpmbuild/SOURCES/containerd.service

